### PR TITLE
SWAP-1251: Add table for units and remove old static unit file

### PR DESCRIPTION
**Pull request by bolmsten**
## Description

A user officer can now create units int the GUI to use in number and interval components.

## Motivation and Context

Allow user officer to change units without need for development effort

## How Has This Been Tested

E2E test

## Fixes

SWAP-1251

## Changes

Changes interval and number component to use units fetched from backend. It also adds a new table for adding and deleting units

## Depends on

https://github.com/UserOfficeProject/user-office-backend/pull/160

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: src/components/MenuItems.tsx [hunk 1]
```diff
@@ -240,6 +240,12 @@ const SettingsMenuListItem = () => {
         <ListItemText primary="Settings" />
       </ListItem>
       <Collapse in={isExpanded} timeout="auto" unmountOnExit>
+        <ListItem component={NavLink} to="/Units" button>
+          <ListItemIcon>
+            <ProposalSettingsIcon />
```
Comment (martin-trajanovski) [inline src/components/MenuItems.tsx:245]: Are we using the same icon for both Proposal statuses and Units?

File: src/components/questionary/questionaryComponents/Interval/QuestionNumberInputForm.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import makeStyles from '@material-ui/core/styles/makeStyles';
```
Comment (martin-trajanovski) [inline src/components/questionary/questionaryComponents/Interval/QuestionNumberInputForm.tsx:1]: Not sure why this `QuestionNumberInputForm` is inside `Interval` folder. I see the same file name inside `NumberInput` form. Or maybe it is not the same but correct me if I am wrong.

File: src/components/settings/unitList/UnitTable.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        data={units}
+        isLoading={loadingUnits}
+        options={{
+          search: true,
+          debounceInterval: 400,
+        }}
+        urlQueryParams={urlQueryParams}
+        setUrlQueryParams={setUrlQueryParams}
+        actions={[
+          rowActionData => {
```
Comment (martin-trajanovski) [inline src/components/settings/unitList/UnitTable.tsx:80]: I think we shouldn't use custom delete functionality. You just need to use existing `SuperMaterialTable` delete property. Something like `delete={deleteUnit}` and define the `deleteUnit` function.

**Review by martin-trajanovski — CHANGES_REQUESTED** [review]

**Review by jekabs-karklins — APPROVED** [review]
Looks good.
One thing I see is that you removed physical property and only left the unit, which is simpler, and also allows to specify units of multiple properties in the same question. but that also raises the question of how will we do a search on that question type. 

Say I want to find all samples where sample weight is greater than 1 kg.

For these types of query, we will need to
1, Add the unit back. We can just have another column in the units table called "property" (easy)
2. For every property, we need a conversion formula to SI units, (medium)
3. Figure out how to tackle the query where multiple physical property units are allowed. (hard)

File: src/components/MenuItems.tsx (hunk 1 above)
Comment (bolmsten) [reply src/components/MenuItems.tsx]: Fixed,

File: src/components/settings/unitList/UnitTable.tsx (hunk 3 above)
Comment (bolmsten) [reply src/components/settings/unitList/UnitTable.tsx]: Done

File: src/components/questionary/questionaryComponents/Interval/QuestionNumberInputForm.tsx (hunk 2 above)
Comment (bolmsten) [reply src/components/questionary/questionaryComponents/Interval/QuestionNumberInputForm.tsx]: removed unused file

**Review by martin-trajanovski — APPROVED** [review]

**Review by RasmiaKulan — APPROVED** [review]

**Pull request merged**